src/Database/Type/TimeType.php

```diff
@@ -16,67 +16,81 @@
  */
 namespace Cake\Database\Type;
 
-use Cake\Chronos\Chronos;
-use Cake\I18n\DateTime;
-use DateTime as NativeDateTime;
-use DateTimeImmutable;
+use Cake\Chronos\ChronosTime;
+use Cake\Database\Driver;
+use Cake\I18n\Time;
 use DateTimeInterface;
-use Exception;
 use InvalidArgumentException;
 
 /**
  * Time type converter.
  *
  * Use to convert time instances to strings & back.
  */
-class TimeType extends DateTimeType
+class TimeType extends BaseType implements BatchCastingInterface
 {
     /**
-     * @inheritDoc
+     * The PHP Time format used when converting to string.
+     *
+     * @var string
      */
     protected string $_format = 'H:i:s';
 
     /**
-     * {@inheritDoc}
+     * Whether `marshal()` should use locale-aware parser with `_localeMarshalFormat`.
+     *
+     * @var bool
+     */
+    protected bool $_useLocaleMarshal = false;
+
+    /**
+     * The locale-aware format `marshal()` uses when `_useLocaleParser` is true.
+     *
+     * See `Cake\I18n\Time::parseTime()` for accepted formats.
+     *
+     * @var string|int|null
+     */
+    protected string|int|null $_localeMarshalFormat = null;
+
+    /**
+     * The classname to use when creating objects.
      *
-     * @var array<string>
+     * @var class-string<\Cake\I18n\Time>
      */
-    protected array $_marshalFormats = [
-        'H:i:s',
-        'H:i',
```

Comment:
Does this mean values in `H:i` format won't be marshalled to `Time()` instance unless local aware parsing is used?

Reply:
No, anything ChronosTime supports is allowed.